File: app/src/main/java/jose/com/bookworm/MainActivity.java
package jose.com.bookworm;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.util.List;

import io.objectbox.Box;
import io.objectbox.BoxStore;
import jose.com.bookworm.model.Book;

public class MainActivity extends AppCompatActivity {

    private TextView textView;
    private EditText titleEditText;
    private EditText authorEditText;
    private Button addBookButton;
    private Button getBookButton;
    private Box<Book> bookBox;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setTitle("BookWorm");
        BoxStore boxStore =((App) getApplication()).getBoxStore();
        bookBox = boxStore.boxFor(Book.class);
        RecyclerView recyclerView = findViewById(R.id.recycler_view);
        recyclerView.hasFixedSize();
        List<Book> allBooksList = bookBox.getAll();
        BookAdapter adapter = new BookAdapter(allBooksList);
        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);

        titleEditText = (EditText) findViewById(R.id.title_edit_text);


        authorEditText = (EditText) findViewById(R.id.author_edit_text);


        addBookButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

            }
        });

        View.OnClickListener listener = new View.OnClickListener() {
            @Override
            public void onClick(View view) {

            }
        };

        addBookButton.setOnClickListener(listener);







    }

    void addBook(Book book) {
        bookBox.put(book);
    }

    public void showAllBooks(View view) {
        Intent intent = new Intent(getApplicationContext(),LibraryActivity.class);
        startActivity(intent);

    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        outState.putInt("key",3);
        outState.putBoolean("true",true);
        super.onSaveInstanceState(outState);




    }
}

File: app/src/main/java/jose/com/bookworm/MainActivity.kt
package jose.com.bookworm

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

import jose.com.bookworm.model.Book
import jose.com.bookworm.adapter.BookAdapter
import jose.com.bookworm.model.BookViewModel

class MainActivity : AppCompatActivity() {
  private val textView: TextView? = null
  private var titleEditText: EditText? = null
  private var authorEditText: EditText? = null
  private val addBookButton: Button? = null
  private val getBookButton: Button? = null
  private val allBooksList: List<Book>? = null
  private lateinit var adapter: BookAdapter

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    title = "BookWorm"

    val recyclerView = findViewById<RecyclerView>(R.id.recycler_view)
    recyclerView.hasFixedSize()

    val model = ViewModelProviders.of(this).get(BookViewModel::class.java)
    model.getBooks().observe(this, Observer<List<Book>> { books ->
        adapter = BookAdapter(books)
    })

    val layoutManager = LinearLayoutManager(this)

    layoutManager.orientation = LinearLayoutManager.VERTICAL
    recyclerView.layoutManager = layoutManager
    recyclerView.adapter = adapter

    titleEditText = findViewById<View>(R.id.title_edit_text) as EditText

    authorEditText = findViewById<View>(R.id.author_edit_text) as EditText

    addBookButton!!.setOnClickListener { }

    val listener = View.OnClickListener { }

    addBookButton.setOnClickListener(listener)
  }

  fun showAllBooks(view: View) { startActivity(intent) }
}

